src/fastfield/serializer/gcd.rs

```diff
@@ -0,0 +1,160 @@
+use fastdivide::DividerU64;
+use gcd::Gcd;
+
+pub const GCD_DEFAULT: u64 = 1;
+
+// Find GCD for iterator of numbers
+pub fn find_gcd(numbers: impl Iterator<Item = u64>) -> Option<u64> {
+    let mut numbers = numbers.filter(|n| *n != 0);
+    let mut gcd = (numbers.next()?).gcd(numbers.next()?);
+    if gcd == 1 {
+        return None;
+    }
+
+    let mut gcd_divider = DividerU64::divide_by(gcd);
+    for val in numbers {
+        let remainder = val - (gcd_divider.divide(val)) * gcd;
+        if remainder == 0 {
+            continue;
+        }
+        gcd = gcd.gcd(val);
+        if gcd == 1 {
+            return None;
```

Comment:
```return Some(1);```

I think the `Option<_>` is mostly nice to return None when there are no element.
Like min for instance.